src/libraries/System.Runtime/tests/System/TypedReferenceTests.cs

```diff
@@ -123,7 +123,7 @@ public static void GetTargetTypeTests()
             Assert.Equal(boolValue.GetType(), TypedReference.GetTargetType(reference));
         }
 
-        [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsAssemblyLoadingSupported))]
```

Comment:
This is an ILC bug. We need to change:

https://github.com/dotnet/runtime/blob/55e2378d86841ec766ee21d5e504d7724c39b53b/src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs#L851

into

```csharp
_dependencies.Add(_factory.MaximallyConstructableType(type), reason);
```